From e2e0180ede2b59ebaa1a0d16e44e8c6b57a5b026 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Fri, 14 Aug 2015 14:25:05 +0200
Subject: [PATCH] Akonadi Contact: don't list Phonon include directory in
 public interface

Phonon is a private dependency, so we must not leak the include paths
to INTERFACE_INCLUDE_DIRECTORIES.
---
 akonadi-contact/src/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/akonadi-contact/src/CMakeLists.txt b/akonadi-contact/src/CMakeLists.txt
index c3324d2..98cbc25 100644
--- a/akonadi-contact/src/CMakeLists.txt
+++ b/akonadi-contact/src/CMakeLists.txt
@@ -124,8 +124,9 @@ generate_export_header(KF5AkonadiContact BASE_NAME akonadi-contact)
 
 add_library(KF5::AkonadiContact ALIAS KF5AkonadiContact)
 
-target_include_directories(KF5AkonadiContact INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/Akonadi/Contact;${Boost_INCLUDE_DIR};${PHONON_INCLUDES}>")
-target_include_directories(KF5AkonadiContact PUBLIC "$<BUILD_INTERFACE:${Akonadi-Contact_SOURCE_DIR}/src;${Akonadi-Contact_BINARY_DIR}/src;${Boost_INCLUDE_DIR};${PHONON_INCLUDES}>")
+target_include_directories(KF5AkonadiContact INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/Akonadi/Contact;${Boost_INCLUDE_DIR}>")
+target_include_directories(KF5AkonadiContact PUBLIC "$<BUILD_INTERFACE:${Akonadi-Contact_SOURCE_DIR}/src;${Akonadi-Contact_BINARY_DIR}/src;${Boost_INCLUDE_DIR}>")
+target_include_directories(KF5AkonadiContact PRIVATE "$<BUILD_INTERFACE:${PHONON_INCLUDES}>")
 # TODO: Remove when kdepimlibs is split (and fix the broken includes in eg. akonadi-calendar)
 target_include_directories(KF5AkonadiContact INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/akonadi/contact>")
 
-- 
2.1.4

